docs/faststore/docs/dynamic-content/overview.mdx

```diff
@@ -0,0 +1,522 @@
+    const products: ProductsExtraData[] = await getProductsFromThirdPartyApi();
+
+    return {
+      data: products,
+    };
+  },
+  namedExtraData: (_: unknown, { name }: { name: string }) => {
+    return {
+      data: `Named extra data: ${name}`,
+    };
+  },
+};
+```
+
+Also, the `src/graphql/thirdParty/resolvers/extraData.ts` file with its field resolvers:
+
+```ts src/graphql/thirdParty/resolvers/extraData.ts
+import { ExtraDataRoot } from "./query";
+
+export const ExtraData = {
+  data: (root: ExtraDataRoot) => root.data,
+  customFieldFromRoot: (root: ExtraDataRoot) => root?.data?.[0]?.image ?? "",
+  customField: async (_: ExtraDataRoot) => {
+    const res = await fetch(
+      "https://fakestoreapi.com/products/category/jewelery"
+    );
+    const customField = await res.json();
+    return (customField?.[0]?.title as string) ?? "";
+  },
+};
+```
+
+Finally, include these resolvers in the `src/graphql/thirdParty/resolvers/index.ts` file:
+
+```ts src/graphql/thirdParty/resolvers/index.ts
+import { ExtraData } from "./extraData";
+import { Query } from "./query";
+
+const resolvers = {
+  ExtraData,
+  Query,
+};
+
+export default resolvers;
+```
+
+With these two new queries created as API Extensions, they are now available to be executed in the `src/dynamicContent/index.ts` file, so use the gql function to parse the query
+
+```ts src/dynamicContent/index.ts
+import { gql } from "@faststore/core/api";
+
+// 2nd approach - Using API Extensions
+// creates a query to fetch the data from API Extensions
+const query = gql(`
+    query ServerDynamicContent($name: String!){
+      extraData {
+        data {
+          title
+          rating {
+            rate
+            count
+          }
+        }
+        customField
+        customFieldFromRoot
+      }
+      namedExtraData(name: $name) {
+        data
+      }
+    }
+`);
+
+// fetches the data using execute function
+async function fetchDataMyLandingPage() {
+  // data fetching not implemented yet
+}
+
+// map the slug to correspondent data-fetching functions
+const dynamicContent = {
+  "my-landing-page": fetchDataMyLandingPage,
+};
+
+export default dynamicContent;
+```
+
+With the query created, run the following command:
+
+```sh
+yarn dev
+```
+
+Now, ensure the GraphQL schema, types, and optimizations will be successfully generated as we can see in the image below:
+
+![GraphQL successfully generated](https://vtexhelp.vtexassets.com/assets/docs/src/graphqlsuccess___d76e7099f3ccf089215101dce9b9cb46.png)
+
+Back to the code, now it's possible to execute the query and leverage the generated types by using the execute method from from the `@faststore/core/experimental` package.
+
+```ts src/dynamicContent/index.ts
+import {
+  ServerDynamicContentQuery,
+  ServerDynamicContentQueryVariables,
+  gql,
+} from "@faststore/core/api";
+import { execute_unstable as execute } from "@faststore/core/experimental";
```

Comment:
@Mariana-Caetano this is a new export that we need to add to the [experimental exports docs list](https://github.com/vtex/faststore/blob/main/apps/site/pages/docs/building-sections/experimental-exports.mdx)